### PR TITLE
[DDW-438] View wallet public key

### PR DESCRIPTION
**Pull request by yakovkaravelov**
This PR created `wallet public key` view section on wallet settings screen
[Jira Ticket](https://jira.iohk.io/browse/DDW-438)

## Todos
- [x] Create `wallet public key` view section on wallet settings screen
- [x] Display public key on the screen with the QR code and copy action @yakovkaravelov 

## Issues
1. Make [focus border](https://user-images.githubusercontent.com/2539345/102903044-d7aeec80-4480-11eb-9c72-1ae191d87c08.png) for tiny inline buttons equal 2px (now 4px) 
2. When we hover left part we show tooltip “Copy” on the button, but it still has unhovered state. It should be “hovered”. 
3. [Red area](https://user-images.githubusercontent.com/2539345/102903152-fc0ac900-4480-11eb-846b-e438312aef77.png) is also clickable and makes “copy” action :scream:

## Testing Checklist
- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1607697377030900)

### App
- [x] Go to wallet settings page and make sure that `wallet public key` view section is existing
- [x] Make sure REVEAL/HIDE toggle works properly
- [x] Click on copy button and make sure toastr pops up indicating the copy of wallet public key
- [x] Click on `show QR code` button and make sure QR code is shown on modal dialog

### Storybook
- [x] Go to `Wallets` => `Settings` => `Wallet Settings` and make sure `wallet public key` view section is existing
- [x] Make sure REVEAL/HIDE toggle works properly
- [x] Go to `Wallets` => `Settings` => `Public Key - QR Code` and make sure QR code is shown properly

## Test Cases

**Scenario 1: View wallet public key field**
Given  I am on wallet Settings page
Then I can view wallet public key field
And the public key is not displayed

**Scenario 2: View public key**
Given  I am on wallet Settings page
And wallet public key field is displayed 
When I select “Reveal” button
Then my wallet public key is shown

**Scenario 3: Copying the wallet public key**
Given that I have revealed my wallet public key
When I select the copy icon
Then wallet public key is copied onto the clipboard

**Scenario 4: Hiding the wallet public key**
Given that I have revealed my wallet public key
When I select Hide” button
Then wallet public key is hidden

**Scenario 5: Show QR Code**
Given that I have revealed my wallet public key
When I select “QR” button
Then a modal dialog opens
And QR Code for the wallet public key is displayed
When I scan the QR Code
Then the correct wallet public key is displayed
When I select ‘Close’
The modal dialog box is dismissed

**Scenario 6: Ensure text is correct**
Given  I am on wallet Settings page
And I am on ‘lang’ language
Then the followings are correct
|lang|English|Japanese|
|--|--|--|
||Wallet public key|ウォレット公開鍵|
||Reveal|表示する|
||Hide|非表示にする|
||Copy|コピーする|
||To show wallet public key click "Reveal" button|ウォレット公開鍵を表示するには[表示]ボタンをクリックしてください|
||Show QR code|QRコードを表示する|
||Public key: {publicKey} copied to clipboard|公開鍵：{publicKey}をクリップボードにコピーしました|

## Testing Summary
- Please refer the snapshot and try testing the scenario described

 For build 15648: (Note: Tested using fake wallet public key)
|Test cases | Results|Note|
|-----------|----|----|
|Scenario 1|Pass||
|Scenario 2|Pass||
|Scenario 3|Pass||
|Scenario 4|Pass||
|Scenario 5|Pass||
|Scenario 6|Pass||

English
image
image
image
image

Japanese
image
image
image
image

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by darko-mijic** [comment]
Screenshot please @yakovkaravelov. Even though we are missing the endpoint.

**Comment by darko-mijic** [comment]
Is the Storybook story implemented @yakovkaravelov?

**Comment by yakovkaravelov** [comment]
> Is the Storybook story implemented @yakovkaravelov?

Yes, they were implemented, but anyway, just added minor update again.
Thanks.

File: source/renderer/app/i18n/locales/ja-JP.json
Comment (DeeJayElly) [inline source/renderer/app/i18n/locales/ja-JP.json:197]: Can you add JP translations?

File: source/renderer/app/i18n/locales/ja-JP.json
Comment (DeeJayElly) [inline source/renderer/app/i18n/locales/ja-JP.json:147]: Can you add JP translations?

File: source/renderer/app/i18n/locales/ja-JP.json
Comment (DeeJayElly) [inline source/renderer/app/i18n/locales/ja-JP.json:127]: Can you add JP translations?

File: source/renderer/app/i18n/locales/ja-JP.json
Comment (DeeJayElly) [inline source/renderer/app/i18n/locales/ja-JP.json:521]: Can you add JP translations?

File: source/renderer/app/stores/WalletsStore.js
Comment (DeeJayElly) [inline source/renderer/app/stores/WalletsStore.js:299]: I assume this is left intentionally until api is ready, right @yakovkaravelov ?

File: source/renderer/app/containers/wallet/WalletSettingsPage.js
Comment (DeeJayElly) [inline source/renderer/app/containers/wallet/WalletSettingsPage.js:45]: Can you change it to 

`const { activePublicKey } = wallets;` and than forward it like:

`walletPublicKey={activePublicKey}`

File: source/renderer/app/containers/wallet/WalletSettingsPage.js
Comment (DeeJayElly) [inline source/renderer/app/containers/wallet/WalletSettingsPage.js:20]: Can we move this constant to some config file like `walletsConfig.js` or `cryptoConfig.js` ?

File: source/renderer/app/containers/wallet/dialogs/settings/WalletPublicKeyQRCodeDialogContainer.js
Comment (DeeJayElly) [inline source/renderer/app/containers/wallet/dialogs/settings/WalletPublicKeyQRCodeDialogContainer.js:18]: You can shorten these 2 lines to 1 line:

`const { active, activePublicKey } = wallets;`

**Review by DeeJayElly — CHANGES_REQUESTED** [review]

File: source/renderer/app/i18n/locales/ja-JP.json
Comment (yakovkaravelov) [reply source/renderer/app/i18n/locales/ja-JP.json:197]: @DeeJayElly This is out of scope of this PR.

File: source/renderer/app/i18n/locales/ja-JP.json
Comment (yakovkaravelov) [reply source/renderer/app/i18n/locales/ja-JP.json:147]: @DeeJayElly This is out of scope of this PR.

File: source/renderer/app/i18n/locales/ja-JP.json
Comment (yakovkaravelov) [reply source/renderer/app/i18n/locales/ja-JP.json:127]: @DeeJayElly This is out of scope of this PR.

File: source/renderer/app/i18n/locales/ja-JP.json
Comment (yakovkaravelov) [reply source/renderer/app/i18n/locales/ja-JP.json:521]: @DeeJayElly This is out of scope of this PR.

File: source/renderer/app/containers/wallet/WalletSettingsPage.js
Comment (yakovkaravelov) [reply source/renderer/app/containers/wallet/WalletSettingsPage.js:45]: @DeeJayElly It is following line 44 naming to make them consistent.

**Comment by yakovkaravelov** [comment]
@nikolaglumac Added config variable to enable this feature. Thanks.

**Comment by yakovkaravelov** [comment]
@DeeJayElly Fixed some of your feedbacks. Thanks.

**Review by DeeJayElly — APPROVED** [review]
Looks ok, gj @yakovkaravelov

**Comment by mioriohk** [comment]
@yakovkaravelov Finding for build 15644. Understandably a few tests failed due to api to get `role` and `index` parameters not ready yet
|Test cases | Results|Note|
|-----------|----|----|
|Scenario 1|Pass||
|Scenario 2|Fail|Upon clicking 'Reveal', no wallet public key displayed|
|Scenario 3|Fail|Unable to copy, no wallet public key displayed|
|Scenario 4|Fail|Unable to hide no wallet public key displayed|
|Scenario 5|Fail|Upon clicking 'QR' code, blank screen displayed|
|Scenario 6|Partial Pass|All text is correct but unable to verify 'Public key: {publicKey} copied to clipboard' as public key not displayed|

**Comment by yakovkaravelov** [comment]
> @yakovkaravelov Finding for build 15644. Understandably a few tests failed due to api to get `role` and `index` parameters not ready yet
> 
> Test cases	Results	Note
> Scenario 1	Pass	
> Scenario 2	Fail	Upon clicking 'Reveal', no wallet public key displayed
> Scenario 3	Fail	Unable to copy, no wallet public key displayed
> Scenario 4	Fail	Unable to hide no wallet public key displayed
> Scenario 5	Fail	Upon clicking 'QR' code, blank screen displayed
> Scenario 6	Partial Pass	All text is correct but unable to verify 'Public key: {publicKey} copied to clipboard' as public key not displayed

@mioriohk There was minor issue with sample key handler. You can test again.
Thanks.

**Comment by mioriohk** [comment]
@yakovkaravelov Finding for build 15648. All passed. Please refer to Test Summary for detailed report. (Note: Tested using fake wallet public key)

**Review by mioriohk — APPROVED** [review]
LGTM 👍 . (Note: API will be incorporated in another PR as advised by @yakovkaravelov )

**Review by nikolaglumac — CHANGES_REQUESTED** [review]
Great work @yakovkaravelov 🎉 

I just have one little request:
truncation
address

We also need to wait for a review from @darko-mijic 🙏

**Comment by nikolaglumac** [comment]
@yakovkaravelov we must not forget to set `WALLET_PUBLIC_KEY_SHARING_ENABLED` to `false` before we merge this PR.

**Comment by yakovkaravelov** [comment]
@nikolaglumac Updated notification message as you asked. I will not forget changing config variable before merge.
@darko-mijic Please review this PR. Once you review, we can merge this.

Thank you.

**Comment by ManusMcCole** [comment]
Hi @yakovkaravelov. Here is my testing summary for build 15660.

**Comment by ManusMcCole** [comment]
Hi @yakovkaravelov. Here is my testing summary for build 15660. 

|Test cases | Results|Note|
|-----------|----|----|
|Scenario 1|Pass||
|Scenario 2|Pass||
|Scenario 3|Pass||
|Scenario 4|Pass||
|Scenario 5|Pass||
|Scenario 6|Pass||

*Issues
The requested change from Nikola to move the Elipses from the end of the private key to the centre has not yet been implemented. See <a href="https://drive.google.com/file/d/1Gqw87uZMSiv47LwcvJ6nTCSrbnPEZS3o/view?usp=sharing
">screenshot</a>

**Comment by yakovkaravelov** [comment]
> Hi @yakovkaravelov. Here is my testing summary for build 15660.
> 
> Test cases	Results	Note
> Scenario 1	Pass	
> Scenario 2	Pass	
> Scenario 3	Pass	
> Scenario 4	Pass	
> Scenario 5	Pass	
> Scenario 6	Pass	
> *Issues
> The requested change from Nikola to move the Elipses from the end of the private key to the centre has not yet been implemented. See [screenshot](https://drive.google.com/file/d/1Gqw87uZMSiv47LwcvJ6nTCSrbnPEZS3o/view?usp=sharing )

@ManusMcCole It was only for copy notification, not for that screenshot you made.

**Comment by nikolaglumac** [comment]
@ManusMcCole I have only asked to add the "..." to the middle of the public key in the notification **only**. Not in the field itself...

**Comment by ManusMcCole** [comment]
> @ManusMcCole I have only asked to add the "..." to the middle of the public key in the notification **only**. Not in the field itself...

Ok. Got it. Thanks 👍

**Review by ManusMcCole — APPROVED** [review]

File: source/renderer/app/stores/WalletsStore.js
Comment (nikolaglumac) [inline source/renderer/app/stores/WalletsStore.js:953]: @yakovkaravelov I would prefer if we would load the wallet public key either when:
A) User navigates to wallet "Settings" tab, or even better
B) User clicks on the "Reveal" button

**Review by nikolaglumac — CHANGES_REQUESTED** [review]
Great work @yakovkaravelov!
Please fix the issue I have noted in the code. Thanks!

File: source/renderer/app/stores/WalletsStore.js
Comment (yakovkaravelov) [reply source/renderer/app/stores/WalletsStore.js]: Fixed.

**Comment by nikolaglumac** [comment]
@gnpf please review this one 🙏

**Comment by gnpf** [comment]
@yakovkaravelov I tested this on build **15709**. The behavior is as expected in all test scenarios. These are some minor findings on this build:

1 - When I hover the input, the "copy" tooltip is displayed. Then, when I click on the input, the address is selected and copied. Check [this video.](https://drive.google.com/file/d/1399g8vykbTAtp0mVbvR-w2aqLhu8-1YX/view?usp=sharing) 
It's not too critical, but think it makes redundant the function of the "Copy" button. Could this be improved so these only reproduce when interacting with "Copy"? 

2 - After clicking "Reveal/Hide" and after closing the QR dialog, there is an outlining on the "Reveal/Hide" and "QR" button respectively. Check [this video](https://drive.google.com/file/d/1DJOgkv8pMCr3pE2p17XlbifgfvGTsVlr/view?usp=sharing) and [this screenshot](https://drive.google.com/file/d/10FjGxtzlug-QiWScpQkqjo6SQpUDgRw8/view?usp=sharing). 
Also, not too important, but maybe this can be improved? Happens on all themes, except Cardano and Light Blue.

**Comment by nikolaglumac** [comment]
> @yakovkaravelov I tested this on build **15709**. The behavior is as expected in all test scenarios. These are some minor findings on this build:
> 
> 1 - When I hover the input, the "copy" tooltip is displayed. Then, when I click on the input, the address is selected and copied. Check [this video.](https://drive.google.com/file/d/1399g8vykbTAtp0mVbvR-w2aqLhu8-1YX/view?usp=sharing)
> It's not too critical, but think it makes redundant the function of the "Copy" button. Could this be improved so these only reproduce when interacting with "Copy"?

This is per design. Not an issue.

> 2 - After clicking "Reveal/Hide" and after closing the QR dialog, there is an outlining on the "Reveal/Hide" and "QR" button respectively. Check [this video](https://drive.google.com/file/d/1DJOgkv8pMCr3pE2p17XlbifgfvGTsVlr/view?usp=sharing) and [this screenshot](https://drive.google.com/file/d/10FjGxtzlug-QiWScpQkqjo6SQpUDgRw8/view?usp=sharing).
> Also, not too important, but maybe this can be improved? Happens on all themes, except Cardano and Light Blue.

@yakovkaravelov please inspect this. I believe buttons are left in “active” state after clicking

**Comment by yakovkaravelov** [comment]
@nikolaglumac Updated things we discussed, thanks.

**Comment by gnpf** [comment]
@yakovkaravelov On build 15737 I can still reproduce this issue, now on all themes:

> 2 - After clicking "Reveal/Hide" and after closing the QR dialog, there is an outlining on the "Reveal/Hide" and "QR" button respectively.

Check [this new video.](https://drive.google.com/file/d/1p3mSAk2NaUSOkh-keVNepeNo7CSWSWxW/view?usp=sharing)

**Comment by a-rukin** [comment]
> @yakovkaravelov On build 15737 I can still reproduce this issue, now on all themes:
> 
> > 2 - After clicking "Reveal/Hide" and after closing the QR dialog, there is an outlining on the "Reveal/Hide" and "QR" button respectively.
> 
> Check [this new video.](https://drive.google.com/file/d/1p3mSAk2NaUSOkh-keVNepeNo7CSWSWxW/view?usp=sharing)

this is focus state (enter/space keyboard press will trigger this action again) 
p.s. you can switch focus with "tab key"

**Comment by nikolaglumac** [comment]
> @yakovkaravelov On build 15737 I can still reproduce this issue, now on all themes:
> 
> > 2 - After clicking "Reveal/Hide" and after closing the QR dialog, there is an outlining on the "Reveal/Hide" and "QR" button respectively.
> 
> Check [this new video.](https://drive.google.com/file/d/1p3mSAk2NaUSOkh-keVNepeNo7CSWSWxW/view?usp=sharing)

@gnpf this is **not an issue**. This is expected and correct behaviour!

**Review by gnpf — APPROVED** [review]

**Review by nikolaglumac — APPROVED** [review]
Great work @yakovkaravelov - proud of you 💪 💯

**Pull request merged**
